posts/2024-07-12_teal_app_development_pharmaverseadam/teal-app-development.qmd

```diff
@@ -0,0 +1,208 @@
+date: "2024-07-12"
+# please do not use any non-default categories.
+# You can find the default categories in the repository README.md
+categories: [Technical, Community]
+# feel free to change the image
+---
+
+<!--------------- typical setup ----------------->
+
+```{r setup, include=FALSE}
+long_slug <- "2024-07-12_teal_app_development_pharmaverseadam"
+# renv::use(lockfile = "renv.lock")
+```
+
+<!--------------- post begins here ----------------->
+
+Every developer must solve two difficult problems when creating a Shiny application (in fact, any application) from the ground up: software architecture and data design. In the world of clinical data analysis, however, much development has been aimed at providing a jump-start approach to creating R/Shiny applications that would take away most of the pain caused by these two problems.
+
+This blog should help you get an idea of how easy it is to get started with the [pharmaverse](https://pharmaverse.org/) ecosystem. We will create an interactive clinical data dashboard using [`{teal}`](https://insightsengineering.github.io/teal/latest-tag/)  and will use [`{pharmaverseadam}`](https://pharmaverse.github.io/pharmaverseadam/) as the data source.
+
+First, you would need to install the two packages. It is recommended to use [`{pak}`](https://pak.r-lib.org/) to take advantage of parallel downloads and builds (note that you can use it together with [`{renv}`](https://rstudio.github.io/renv/articles/renv.html) for locking dependencies).
+
+``` r
+pak::pkg_install(c("teal", "pharmaverseadam"))
+```
+
+## **Building a Simple Teal App**
+
+Next, we would create a simple app - basically following [the official teal guide](https://insightsengineering.github.io/teal/latest-tag/articles/getting-started-with-teal.html), but we will use datasets from the {pharmaverseadam} package.
+
+``` r
+# Step 1: import packages
+library(teal)
+library(pharmaverseadam)
+
+# Step 2: create a teal data object
+data <- cdisc_data(
+  ADAE = pharmaverseadam::adae,
+  ADSL = pharmaverseadam::adsl
+)
+
+# Step 3: initialize teal app
+app <- init(
+  data = data,
+  modules = example_module()
+)
+
+# Step 4: run shiny app
+shinyApp(app$ui, app$server)
+```
+
+### **Let’s Take a Closer Look at Each Step:**
+
+-   **At step 1**, we import libraries to use functions and datasets exported by those.
+
+-   **In step 2**, we create a teal data object that can be used in a teal app. There are a couple of subtle details here: instead of the default teal_data() function, we use a special wrapper around it, designed specifically for clinical trial data - cdisc_data(). Its advantage is that it will automatically generate join_keys() for the datasets that we pass in. Please note, that the dataset names should be all caps. The resulting data object is an s4 class instance, and we can verify that the output is correct by checking data names and join_keys slots:
```

Comment:
Use backticks for functions here - and perhaps link to them as well using the `link` package if that works?

Reply:
all done!